### ISSUE TITLE
Fully qualified symbol in def referring to current namespace

### ISSUE THREAD

**Issue by jimka2001** [clj-kondo/clj-kondo#1326]
**version**

I am using the clojure-lsp plugin + lisp-mode for emacs.

**platform**
MacOS

**editor**

emacs

**problem**

I see several wrong errors and warnings.
When I use `alias` to make an alias to the current package, and thereafter
prefix symbols with that alias, the analyser doesn't seem to recognize the names.

![Screenshot 2021-07-24 at 13 47 22](https://user-images.githubusercontent.com/6414129/126867462-de679213-2317-47da-ab7f-7d6c75709499.png)
**repro**

```
(ns jimka-test)
(alias 'jt 'jimka-test)
(def jt/xyz (fn [x] x))
(defn foo [x]
  (jt/xyz x))
(defn bar [x]
  (xyz x))
```

**expected behavior**

I expect that `jt/xyz` be treated like `xyz` if `jt` is aliased to the current namespace.

**Comment by jimka2001** [comment]
As I understand Michiel Borkent already has a workaround for this issue.

**Comment by borkdude** [comment]
A workaround using hooks:

def_hook.clj:
``` clojure
(ns def-hook
  (:require [clj-kondo.hooks-api :as api]))

(defn transform-def [{:keys [:node]}]
  (let [[name-node & arg-nodes] (rest (:children node))
        name-sym (api/sexpr name-node)]
    (when-not (simple-symbol? name-sym)
      (let [new-node (with-meta
                       (api/list-node
                        (list*
                         (api/token-node 'def)
                         (api/token-node (symbol (name name-sym)))
                         arg-nodes))
                       (meta node))]
        {:node new-node}))))
```

``` clojure
{:hooks {:analyze-call {clojure.core/def def-hook/transform-def}}}

```

**Comment by borkdude** [comment]
@jimka2001 I decided to start supporting this one so we can get this out of the way while iterating on the macroexpand stuff.
